### DISCUSSION TITLE
Fake http/tls/etc payload syntax

### DISCUSSION THREAD

**Discussion by b1ackbeat** [bol-van/zapret#278] [Q&A]
Hello, friend! I want to replace www.iana.org at fake ClientHello message with my own domain, but I can not understand the syntax of file which should contain the payload, could you please help to clarity it and give some examples? I mean options dpi-desync-fake-http, dpi-desync-fake-tls and so on.

**Comment by bol-van** [comment]
plain http request you can write in text editor
there's example in files/fake
tls client hello you must sniff and save in wireshark
only tcp data without header
this payload must not contain kyber to fit into single packet
it means do not use chrome. use ff or curl

**Comment by hamedsbt** [comment]
Hi dear @bol-van, Could you tell me if I generate "tls client hello" correctly:
``sudo tcpdump -i enp6s0 -w tls_clienthello.pcap``
``openssl s_client -connect vid.faradars.org:443``

Then open tls_clienthello.pcap file with wireshark and export the "TLSv1.2 Record..." as bin file:
![image](https://github.com/user-attachments/assets/c0254d0f-c69a-450a-be35-abe1c8fdf227)

**Reply by bol-van** [reply]
TLS must start with 16 03 bytes.
There's easier way

nc -l -p 1111 >tls.bin
curl --connect-to ::127.0.0.1:1111 https://iana.org
